src/device/index.js

```diff
@@ -191,10 +196,13 @@ const device = (base, wsBase) => {
     *  console.log('Unsubscribed!');
     * });
     **/
-    unsubscribe: (deviceId, componentId) => axios({
-      method: 'DELETE',
-      url: `${base}/${deviceId}/${componentId}/subscribe`
-    })
+    unsubscribe: (deviceId, componentId) => {
+      var id = deviceId+componentId;
+      if (websockets.hasOwnProperty(id)) {
+        websockets[id].close();
+        delete websockets[id];
```

Comment:
You should return a promise, as the api was previously `thenable`. You can do something like:

```
try {
    websockets[id].close();
    delete websockets[id];
     return Promise.resolve();
} catch (err) {
    return Promise.reject(err);
}
```

Reply:
Oh and do we no longer need to do a `DELETE` on `unsubscribe`? If not we should update the api docs.

Reply:
Oh also I see one line `190` the example doc is wrong. `agile.device.get` should be `agile.device.unsubscribe` Could you update with this PR please?

Reply:
subscribe and unsubscribe goes in pairs. You either use the HTTP version (and then a generic WS to get the messages), or the component specific WS. In this case, you subscribe by opening the WS, and unsubscribe by closing the WS. Note that there could be several WSs subscribed at the same time, and in the bacground at the protocol level we only unsubscribe when all are closed, and also eventual HTTP subscribes had been deleted.

@craig, can you implement the changes you've proposed? Thanks